### DISCUSSION TITLE
Unable to render text with Pango on Linux

### DISCUSSION THREAD

**Discussion by gunchleoc** [ImageMagick/ImageMagick#1989] [General]
Hi all,

I am trying to use Pango for rendering text with Imagemagick, but it is not recognized. I'm on Linux Mint 19.9 Tessa 64 Bit.

ETA: I just tried this again with an Ubuntu Trusty (14.04) VM, and it works without a problem there. This might still be worth looking into though in case it's a bug.

I am trying to run one of the examples from: https://www.imagemagick.org/Usage/text/#pango

```
$ convert -background lightblue pango:"Anthony Thyssen" pango.gif
```
Resulting in:

```
convert-im6.q16: unable to open image `pango:Anthony Thyssen': No such file or directory @ error/blob.c/OpenBlob/2701.
convert-im6.q16: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert-im6.q16: no images defined `pango.gif' @ error/convert.c/ConvertImageCommand/3258.
```

Any pointers will be greatly appreciated :)

My ImageMagick version:

```
$ convert --version
Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org
Copyright: © 1999-2017 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff wmf x xml zlib
```

I checked the available formats with `convert -list format` and Pango is indeed not listed.

Is there a specific Pango package that I am missing?

Here's the ones I have installed:

```
$ apt list --installed | grep pango

gir1.2-coglpango-1.0/bionic-updates,now 1.22.2-3ubuntu1 amd64 [installed]
gir1.2-pango-1.0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libcogl-pango20/bionic-updates,now 1.22.2-3ubuntu1 amd64 [installed]
libpango-1.0-0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libpango-perl/bionic,now 1.227-2build1 amd64 [installed]
libpango1.0-0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libpango1.0-dev/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libpangocairo-1.0-0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libpangoft2-1.0-0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
libpangomm-1.4-1v5/bionic,now 2.40.1-4 amd64 [installed]
libpangox-1.0-0/bionic,now 0.0.2-5 amd64 [installed,automatic]
libpangoxft-1.0-0/bionic-updates,bionic-security,now 1.40.14-1ubuntu0.1 amd64 [installed]
```

Packages with "pango" in the name that are currently not installed:

```
libcogl-pango-dev/bionic-updates 1.22.2-3ubuntu1 amd64
libghc-pango-dev/bionic 0.13.4.0-1build2 amd64
libghc-pango-doc/bionic,bionic 0.13.4.0-1build2 all
libghc-pango-prof/bionic 0.13.4.0-1build2 amd64
libghc-yi-frontend-pango-dev/bionic 0.13.7-1build3 amd64
libghc-yi-frontend-pango-doc/bionic,bionic 0.13.7-1build3 all
libghc-yi-frontend-pango-prof/bionic 0.13.7-1build3 amd64
libgtk2-ex-entry-pango-perl/bionic,bionic 0.09-1 all
libpango-perl/bionic,now 1.227-2build1 amd64 [installed]
libpango1.0-doc/bionic-updates,bionic-updates,bionic-security,bionic-security 1.40.14-1ubuntu0.1 all
libpango3.0-cil/bionic 2.99.3-2 amd64
libpangomm-1.4-dev/bionic 2.40.1-4 amd64
libpangomm-1.4-doc/bionic,bionic 2.40.1-4 all
libpangox-1.0-dev/bionic 0.0.2-5 amd64
libsdl-pango-dev/bionic 0.1.2-7 amd64
libsdl-pango1/bionic 0.1.2-7 amd64
pango1.0-tests/bionic-updates,bionic-security 1.40.14-1ubuntu0.1 amd64
pangoterm/bionic 0~bzr607-1 amd64
ruby-pango/bionic 3.2.4-1 amd64
```

Please let me know if you need any more info.

**Comment by snibgo** [comment]
Your version includes pangocairo, which is the only built-in delegate that I have, and pango works fine for me.

Your command is fine.

So I don't know what the problem is, or how it might be fixed. All I can suggest is that you try a current release, eg 6.9.11 or 7.0.10.

**Comment by snibgo** [comment]
Something else to try: "magick -list format". For the pango line, I get:
`PANGO* PANGO     r--   Pango Markup Language (Pangocairo 1.42.4)`
If your library is older, you might try updating it and rebuilding IM.

**Reply by gunchleoc** [reply]
The "magick" command does not exist on my system. I did run `convert -list format` which does not find my Pango.

It's interesting that it works on an older Ubuntu with

```
$ convert --version
Version: ImageMagick 6.7.7-10 2018-09-28 Q16 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2012 ImageMagick Studio LLC
Features: OpenMP 
```

There are no delegates listed there, maybe one of those is messing things up on my Mint?

I'll try compiling from source next.

**Reply by gunchleoc** [reply]
No luck, I get:

```
magick --version
magick: error while loading shared libraries: libMagickCore-7.Q16HDRI.so.7: cannot open shared object file: No such file or directory
:~/sources/ImageMagick-7.0.10-0 $
```

Same problem with https://imagemagick.org/download/ImageMagick-6.9.11-10.tar.gz

I'll just stick with my virtual machine for now where it works. Thank you for your time!

Since I'm on a legacy version, feel free to close this and spend your energy on the new version.

For completeness, log output from my build attempt:

[imagemagick-7-buildlog.txt](https://github.com/ImageMagick/ImageMagick/files/4585120/imagemagick-7-buildlog.txt)
